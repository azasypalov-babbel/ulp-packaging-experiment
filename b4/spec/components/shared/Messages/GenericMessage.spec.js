import React from 'react';
import { shallow } from 'enzyme';
import { GenericMessage } from
  '../../../../src/components/shared/Messages/GenericMessage';


describe('<GenericMessage />', () => {
  const props = {
    navigationService: {
      reload: jest.fn()
    },
    navigateToReturnUrl: jest.fn(),
    translations: {
      reload: {
        title: 'generic_error_screen.reload.title',
        body: 'generic_error_screen.reload.body',
        cta: 'generic_error_screen.reload.cta'
      },
      return: {
        title: 'generic_error_screen.return.title',
        body: 'generic_error_screen.return.body',
        cta: 'generic_error_screen.return.cta'
      }
    }
  };

  test('renders', () => {
    const wrapper = shallow(<GenericMessage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when passing only requiered props', () => {
    const wrapper = shallow(<GenericMessage {...props} />);

    describe('when clicking on CTA button', () => {
      beforeEach(() => {
        const ctaButton = wrapper.find('[data-selector="cta-button"]');
        ctaButton.simulate('click');
      });

      test('user is redirected', () => {
        expect(props.navigateToReturnUrl).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe('when passing isReload prop', () => {
    const wrapper = shallow(<GenericMessage {...props} isReload />);

    describe('when clicking on CTA button', () => {
      beforeEach(() => {
        const ctaButton = wrapper.find('[data-selector="cta-button"]');
        ctaButton.simulate('click');
      });

      test('page gets reloaded', () => {
        expect(props.navigationService.reload).toHaveBeenCalledTimes(1);
      });
    });
  });
});
