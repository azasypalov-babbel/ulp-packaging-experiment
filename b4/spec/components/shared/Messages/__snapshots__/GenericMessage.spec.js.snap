// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenericMessage /> renders 1`] = `
<ErrorOverlay
  body="generic_error_screen.return.body"
  dataSelector="error-overlay-generic-error"
  icon={
    <WithTheme(GenericErrorIcon)
      size="3.5rem"
    />
  }
  key="generic-error"
  title="generic_error_screen.return.title"
>
  <WithKeypress(styled.button)
    data-selector="cta-button"
    onClick={[Function]}
  >
    generic_error_screen.return.cta
  </WithKeypress(styled.button)>
</ErrorOverlay>
`;
